edk2toolext/omnicache.py

```diff
@@ -10,252 +10,366 @@
 import argparse
 import datetime
 import yaml
+import uuid
 from io import StringIO
 
 from edk2toolext import edk2_logging
-from edk2toollib import utility_functions
-from edk2toolext.edk2_git import Repo
-
-
-class OmniCacheConfig():
-    '''
-    class to manage the Internal Omnicache config file.
-    Load, Save, Version check, etc.
-    '''
+from edk2toollib.utility_functions import RunCmd
+
+# Omnicache version 0.11 design notes:
+# 1. Only a single remote per URL will be permitted.
+# 2. Remote names inside omnicache will be UUIDs generated when a new URL is added.
+# 3. Users may specify names for remotes, but they will be treated simply as "display names" without git significance.
+# 4. As many objects as possible will be cached for any given remote; so all remotes will fetch tags.
+# 4a. To avoid tag name collisions between remotes, tags will be fetched into per-remote namespaces.
+# 5. Older omnicaches will be updated to meet the above design points when first encountered by the script.
+# 5a. This implementation takes care that the underlying git construction is compatible (but not performant) when an
+#     older omnicache interacts with it. Older omnicache should not crash when encountering a newer omnicache directory
+#     but may take a lot of slow and unecessary actions to "re-initialize" it.
+
+OMNICACHE_VERSION = "0.11"
+
+# Prior to version 0.11, an "omnicache.yaml" file in the root of the omnicache repo was used to manage the cache
+# configuration. Starting with 0.11, git config entries are used directly. If the file below is present, it will be
+# removed to force re-initialization of the omnicache by older omnicache instances. (e.g. if an old branch with a prior
+# implementation of omnicache is being built and uses the same cache directory)
+OMNICACHE_FILENAME = "omnicache.yaml"
```

Comment:
Should we change this CONSTANT name? To maybe PRE_0_11_OMNICACHE_FILENAME or something?

Reply:
Sure - updated.